frontend/src/components/pages/CourseDashboard.js

```diff
@@ -24,21 +69,95 @@ export function CourseDashboard() {
             .catch((error) => setError(error.message));
     }, []);
 
+    useEffect(() => {
+        const userId = Cookies.get('userId');
+
+        if (!userId) {
+            setError('User ID not found');
+            return;
+        }
+
+        fetch(`http://localhost:4000/user/${userId}`)
+            .then((response) => {
+                if (!response.ok) {
+                    throw new Error('Network response was not ok');
+                }
+                return response.json();
+            })
+            .then((data) => setUserInfo(data.user))
+            .catch((error) => setError(error.message));
+    }, []);
+
     if (error) return <p>Error: {error}</p>;
     if (!courseInfo) return <p>Loading...</p>;
+    if (!userInfo) return <p>Loading...</p>;
 
     var courseList = [];
     //Push all courses into courseList
-    courseInfo.forEach(course => {
+    if (userInfo.role === "educator"){
+        courseInfo.forEach(course => {
+            courseList.push(
+                <a href={`/courses/${course.ID}`}>
+                    <div className="bg-gray-100 p-4 rounded shadow hover:bg-gray-300" >
+                        <h3 className="text-xl font-semibold" >{course.title}</h3>
+                        <p className="mt-2">{course.description}</p>
+                    </div>
+                </a>
+            )
+        });
+        //Push the create course button at the end of courses
         courseList.push(
-            <a href={`/courses/${course.ID}`}>
-                <div className="bg-gray-100 p-4 rounded shadow hover:bg-gray-300" >
-                    <h3 className="text-xl font-semibold" >{course.title}</h3>
-                    <p className="mt-2">{course.description}</p>
-                </div>
-            </a>
+            <Popup trigger={<div className="bg-blue-500 p-4 rounded shadow hover:bg-blue-700 m-auto text-center text-xl text-white font-semibold hover:cursor-pointer" > 
+                <div>+ New Course</div> </div>} modal nested>
```

Comment:
Maybe the button should be elsewhere. It looks nice on mobile, but on desktop, having the button after the courses looks a little weird. Maybe having it at the top like here (smaller than the current one)
![Screenshot 2024-11-17 at 10-40-49 React App](https://github.com/user-attachments/assets/ddd0462c-6c44-40f2-aa94-a6c047dc81c0)
![Screenshot 2024-11-17 at 10-44-35 React App](https://github.com/user-attachments/assets/0a3f3473-f2c1-4d98-bfd0-973260161556)

Reply:
Good idea. I went ahead and added this